pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -1878,3 +1878,103 @@ func assertName(t *testing.T, expected, username string) {
 	assert.Regexp(t, dnsRegExp, transformUsername(username))
 	assert.Equal(t, expected, transformUsername(username))
 }
+
+// Test the scenario where the existing usersignup CompliantUsername becomes outdated eg. transformUsername func is changed
+func TestChangedCompliantUsername(t *testing.T) {
+	email := "foo@redhat.com"
+	// starting with a UserSignup that exists and was approved and has the now outdated CompliantUsername
+	userSignup := &v1alpha1.UserSignup{
+		ObjectMeta: newObjectMeta("", email),
+		Spec: v1alpha1.UserSignupSpec{
+			Username:      email,
+			Approved:      true,
+			TargetCluster: "east",
+		},
+		Status: v1alpha1.UserSignupStatus{
+			Conditions: []toolchainv1alpha1.Condition{
+				{
+					Type:   toolchainv1alpha1.UserSignupApproved,
+					Status: v1.ConditionTrue,
+					Reason: toolchainv1alpha1.UserSignupApprovedByAdminReason,
+				},
+				{
+					Status: v1.ConditionTrue,
+					Type:   toolchainv1alpha1.UserSignupComplete,
+				},
+			},
+			CompliantUsername: "foo-old",
+		},
+	}
+
+	// also starting with the old MUR whose name matches the outdated UserSignup CompliantUsername
+	oldMur := &v1alpha1.MasterUserRecord{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo-old",
+			Namespace: operatorNamespace,
+			Labels:    map[string]string{v1alpha1.MasterUserRecordUserIDLabelKey: userSignup.Name},
+		},
+	}
+
+	// create the initial resources
+	r, req, _ := prepareReconcile(t, userSignup.Name, userSignup, oldMur, basicNSTemplateTier)
+
+	// 1st reconcile should effectively be a no op because the MUR name and UserSignup CompliantUsername match and status is all good
+	res, err := r.Reconcile(req)
+	require.NoError(t, err)
+	require.Equal(t, reconcile.Result{}, res)
+
+	// after the 1st reconcile verify that the MUR still exists and its name still matches the initial UserSignup CompliantUsername
+	murs := &v1alpha1.MasterUserRecordList{}
+	err = r.client.List(context.TODO(), murs)
+	require.NoError(t, err)
+	require.Len(t, murs.Items, 1)
+	mur := murs.Items[0]
+	require.Equal(t, operatorNamespace, mur.Namespace)
```

Comment:
you can list MURs in a specific namespace, then you wouldn't need to verify the namespace in the retrieved resource

```suggestion
	err = r.client.List(context.TODO(), murs, client.InNamespace(operatorNamespace))
	require.NoError(t, err)
	require.Len(t, murs.Items, 1)
	mur := murs.Items[0]
```

Reply:
Good point, thanks!